tests/test_core/test_storage.py

```diff
@@ -68,6 +68,36 @@ def test_get_custom_folder(local_storage_manager: StorageManager, project_folder
     assert local_storage_manager.get_folder("eopatches", full_path=True) == abs_path
 
 
+@pytest.mark.parametrize("config_profile", [None, "", "nonexistent-config-profile"])
+@pytest.mark.parametrize("env_profile", [None, "", "nonexistent-env-profile"])
+def test_aws_profile(aws_storage_config: RawConfig, config_profile: Optional[str], env_profile: Optional[str]):
+    """Checks different combinations of profile being set with a config parameter and environmental variable. Checks
+    also that config parameter takes priority over environmental variable."""
+
+    for parameter_key, parameter_value, config_dict in [
+        ("aws_profile", config_profile, aws_storage_config),
+        ("AWS_PROFILE", env_profile, os.environ),
+    ]:
+        if parameter_value is not None:
+            config_dict[parameter_key] = parameter_value
+        elif parameter_key in config_dict:
+            del config_dict[parameter_key]
```

Comment:
The setup step is not straightforward to understand (the idea of what we are trying to accomplish here), some more info in the description would be great for maintainability

Reply:
Added a couple of sentences of extra explanation into the docstring of the function. Should be clearer now.